### PR TITLE
Use modern test tools

### PR DESCRIPTION
**Pull request by andreasnoack**
Drop 0.4 support

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,43 +1,22 @@
...
+    info("$(typeof(A))  $(size(A))")
 
-        # verify mmread(mmwrite(A)) == A
-        MatrixMarket.mmwrite(new_filename, A)
-        new_A = MatrixMarket.mmread(new_filename)
-        assert(new_A == A)
+    # verify mmread(mmwrite(A)) == A
+    MatrixMarket.mmwrite(new_filename, A)
+    new_A = MatrixMarket.mmread(new_filename)
+    @test new_A != A
```
Comment (tkonolige) [inline test/runtests.jl]: This should be `@test new_A == A`

**Comment by tkonolige** [comment]
Is there a reason for dropping 0.4 support?

File: test/runtests.jl (hunk 1 above)
Comment (andreasnoack) [inline test/runtests.jl]: Ha. I did that to check the behavior when it failed. I'll update.

**Comment by andreasnoack** [comment]
> Is there a reason for dropping 0.4 support?

Otherwise, we'd have to add a dependency on `TestBaseNext`. The existing version of the package will continue to work on 0.4 and 0.4 will also soon be forgotten so I thought it was better to drop support than add a dependency.

**Pull request merged**